crates/nu-command/src/filters/sort.rs

```diff
@@ -220,22 +221,22 @@ fn sort_record(
             b.0.clone()
         };
 
-        // Convert to lowercase if case-insensitive
+        // Fold case if case-insensitive
         let left = if insensitive {
-            left_res.to_ascii_lowercase()
+            left_res.to_folded_case()
         } else {
             left_res
         };
         let right = if insensitive {
-            right_res.to_ascii_lowercase()
+            right_res.to_folded_case()
         } else {
             right_res
         };
 
         if natural {
             compare_str(left, right)
         } else {
-            left.partial_cmp(&right).unwrap_or(Ordering::Equal)
+            left.cmp(&right)
```

Comment:
Going for the correct `.cmp` shouldn't introduce any changes as `str: Ord + PartialOrd` the whole time, right?

Reply:
Correct. `.partial_cmp` won't (shouldn't) ever return `None` for (correctly implemented) `Ord` types.